share/shwap/sample_test.go

```diff
@@ -54,55 +36,53 @@ func TestSampleValidate(t *testing.T) {
 // TestSampleNegativeVerifyInclusion checks
 func TestSampleNegativeVerifyInclusion(t *testing.T) {
 	const odsSize = 8
-	eds := edstest.RandEDS(t, odsSize)
-	root, err := share.NewRoot(eds)
+	randEDS := edstest.RandEDS(t, odsSize)
+	root, err := share.NewRoot(randEDS)
 	require.NoError(t, err)
+	inMem := eds.Rsmt2D{ExtendedDataSquare: randEDS}
 
-	sample, err := SampleFromEDS(eds, rsmt2d.Row, 0, 0)
+	sample, err := inMem.Sample(context.TODO(), 0, 0)
+	require.NoError(t, err)
+	err = sample.Validate(root, 0, 0)
 	require.NoError(t, err)
-	included := sample.verifyInclusion(root, 0, 0)
-	require.True(t, included)
 
 	// incorrect row index
-	included = sample.verifyInclusion(root, 1, 0)
-	require.False(t, included)
-
-	// incorrect col index is not used in the inclusion proof verification
-	included = sample.verifyInclusion(root, 0, 1)
-	require.True(t, included)
+	err = sample.Validate(root, 1, 0)
+	require.ErrorIs(t, err, shwap.ErrorFailedVerification)
 
 	// Corrupt the share
 	sample.Share[0] ^= 0xFF
-	included = sample.verifyInclusion(root, 0, 0)
-	require.False(t, included)
+	err = sample.Validate(root, 0, 0)
+	require.ErrorIs(t, err, shwap.ErrorFailedVerification)
 
 	// incorrect proofType
-	sample, err = SampleFromEDS(eds, rsmt2d.Row, 0, 0)
+	sample, err = inMem.Sample(context.TODO(), 0, 0)
 	require.NoError(t, err)
 	sample.ProofType = rsmt2d.Col
-	included = sample.verifyInclusion(root, 0, 0)
-	require.False(t, included)
+	err = sample.Validate(root, 0, 0)
+	require.ErrorIs(t, err, shwap.ErrorFailedVerification)
 
 	// Corrupt the last root hash byte
-	sample, err = SampleFromEDS(eds, rsmt2d.Row, 0, 0)
+	sample, err = inMem.Sample(context.TODO(), 0, 0)
```

Comment:
TODO hints that a real context should be added sometime in the future, and it was added during the times when std started adopting context, but in this case, we should just pas Background